mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs

```diff
@@ -263,9 +255,18 @@ void ConstructCRef (StringBuilder sb)
 			if (DescKind == Kind.Type)
 				return;
 
+			if (ExplicitImplMember != null) {
+				sb.Append ('$');
```

Comment:
This is also why nested types use `+` instead of `.`, to facilitate a simple `string.Split()`-based parsing strategy. Consider:

    var parts = "A.B.C.D.E".Split(".");

For *members* (`F` fields, `M` methods, `P` properties, etc.) the *last* element in the array is the member name, the second-to-last element is the type name -- including nested types -- and everything prior to that is part of the namespace.

For *types*, the last element of the array is the typename, and everything preceding is the namespace.

For *namespaces*, the entire array is the namespace.

This is why all CREFs require a prefix, so you know how to interpret the array. `System.Object` is meaningless; `T:System.Object` can be parsed.